libraries/amqp/reliable_amqp_publisher.cpp

```diff
@@ -182,6 +193,10 @@ void reliable_amqp_publisher_impl::pump_queue() {
 bool reliable_amqp_publisher_impl::verify_max_queue_size() {
    constexpr unsigned max_queued_messages = 1u << 20u;
 
+   if(message_deque.size() > max_queued_messages - 1000) {
```

Comment:
No more arbitrary than `1u << 20u`. The idea is that since most implementations will call `app().quit()` the message will still be queued before hitting `max_queued_messages` and application is shutdown. If this was called when `message_deque.size() > max_queued_messages` then we would not queue/save the current message.

Since in all cases, `on_fatal_error` calls `app().quit()`, I suggest I change this to never stop queueing but only call on_fatal_error. @spoonincode do you have any issue with that?

```
   if(message_deque.size() > max_queued_messages) {
      if(logged_exceeded_max_depth == false)
         elog("AMQP connection ${a} publishing to \"${e}\" has reached ${max} unconfirmed messages",
              ("a", retrying_connection.address())("e", exchange)("max", max_queued_messages));
      //logged_exceeded_max_depth = true; // will not need this
      std::string err = "AMQP publishing to " + exchange + " has reached " + std::to_string(message_deque.size()) + " unconfirmed messages";
      on_fatal_error(err);
      return true; // logically, will just change this method to be void
   }
```

Reply:
Seems fine to me.